azext_edge/edge/providers/rpsaas/adr/namespaces.py

```diff
@@ -0,0 +1,236 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from rich.console import Console
+from typing import TYPE_CHECKING, Dict, List, Iterable, Optional
+from knack.log import get_logger
+
+from ....common import IdentityType
+from ....util.az_client import get_registry_refresh_mgmt_client, get_resource_client, wait_for_terminal_state
+from ....util.common import parse_kvp_nargs
+from ....util.queryable import Queryable
+
+if TYPE_CHECKING:
+    from ....vendor.clients.deviceregistryrefreshmgmt.operations import NamespacesOperations
+    from ....vendor.clients.resourcesmgmt.operations import ResourcesOperations
+
+
+console = Console()
+logger = get_logger(__name__)
+EVENTGRIDTOPIC_API_VERSION = "2025-02-15"
+NAMESPACE_RESOURCE_TYPE = "Microsoft.DeviceRegistry/namespaces"
+
+
+class Namespaces(Queryable):
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.deviceregistry_mgmt_client = get_registry_refresh_mgmt_client(
+            subscription_id=self.default_subscription_id
+        )
+        self.resource_mgmt_client = get_resource_client(
+            subscription_id=self.default_subscription_id
+        )
+        self.ops: "NamespacesOperations" = self.deviceregistry_mgmt_client.namespaces
+        self.resource_ops: "ResourcesOperations" = self.resource_mgmt_client.resources
+
+    def create(
```

Comment:
question: `create` and `add_endpoint` are basically calling same api but modify different part of the properties?

Reply:
create creates the initial endpoints, add/remove endpoint is supposed to do the post create update endpoints

I see that I did not do patch for the update + add/remove endpoint commands - going to fix that